File: snapshot_profile.py
from playwright.sync_api import sync_playwright

with sync_playwright() as p:
    browser = p.chromium.launch(headless=True)
    page = browser.new_page()
    page.goto('http://localhost:8502')
    
    print("Logging in...")
    page.fill('input[placeholder="Enter username"]', 'manager')
    page.fill('input[placeholder="Enter password"]', 'manager123')
    page.click('button:has-text("Sign in →")')
    
    print("Waiting for dashboard...")
    page.wait_for_selector('text=Operations Dashboard', timeout=10000)
    page.wait_for_timeout(2000)
    
    print("Opening popover...")
    page.click('button:has-text("👤 M")')
    
    print("Capturing screenshot...")
    page.wait_for_selector('text=manager@ibm.com', timeout=5000)
    page.wait_for_timeout(1000)
    
    page.screenshot(path='profile_popover.png')
    print("Done!")
    
    browser.close()

File: snapshot_profile.js
const { chromium } = require('playwright');

(async () => {
  const browser = await chromium.launch({ headless: true });
  const page = await browser.newPage();
  
  console.log("Navigating to app...");
  await page.goto('http://localhost:8502');
  
  console.log("Logging in as manager...");
  await page.fill('input[placeholder="Enter username"]', 'manager');
  await page.fill('input[placeholder="Enter password"]', 'manager123');
  await page.click('button:has-text("Sign in →")');
  
  // Wait for dashboard to load
  await page.waitForSelector('text=Operations Dashboard', { timeout: 10000 });
  await page.waitForTimeout(2000); // let charts render
  
  console.log("Opening profile popover...");
  // Find the button with the User Profile emoji "👤 M"
  await page.click('button:has-text("👤 M")');
  
  // Wait for the popover content to appear
  await page.waitForSelector('text=manager@ibm.com', { timeout: 5000 });
  await page.waitForTimeout(1000);
  
  await page.screenshot({ path: 'profile_popover.png' });
  console.log("Screenshot saved to profile_popover.png");
  
  await browser.close();
})();
